src/python/numpy_helper.hpp

```diff
@@ -61,11 +61,7 @@ template <> struct numpy_type<unsigned long>  { static const long id = NPY_ULONG
 #endif
 
 inline PyObject *as_python_string(const std::string &in) {
-#if PY_MAJOR_VERSION < 3
-  return PyString_FromStringAndSize(in.c_str(), in.size());
-#else
   return PyUnicode_FromStringAndSize(in.c_str(), in.size());
-#endif
```

Comment:
Hmm, even in Python 2 (I guess 2.x is becoming less and less important...)? It might make sense to have an option for Py3 that always returns PyBytes, too. IIRC pandas is more Python 2-centric, i.e. bytes are returned unless an encoding is passed, in which case you get unicode:

https://github.com/pydata/pandas/blob/master/pandas/parser.pyx#L1972